paramiko/agent.py

```diff
@@ -392,11 +401,13 @@ class AgentKey(PKey):
     work as expected.
     """
 
-    def __init__(self, agent, blob):
+    def __init__(self, agent, blob, rsa_prefs):
         self.agent = agent
         self.blob = blob
         self.public_blob = None
         self.name = Message(blob).get_text()
+        if(self.name in rsa_prefs):
+            self.name = rsa_prefs[0]
```

Comment:
Not sure how I should negotiate the key. This is (I think) only taking the clients' preference into consideration, and will fail if the client prefers SHA2 and the server does not support SHA2.

https://datatracker.ietf.org/doc/html/draft-ietf-curdle-rsa-sha2-12#section-3.3